src/PeliasSearchManager.swift

```diff
@@ -157,16 +157,29 @@ open class PeliasResponse: APIResponse {
   /// The url response if the request completed successfully.
   open let response: URLResponse?
   /// The error if an error occured executing the operation.
-  open let error: NSError?
+  open var error: NSError?
   ///
   open var parsedResponse: PeliasSearchResponse?
-  
+
+  private static let validTypes: Set = ["Point", "MultiPoint", "LineString", "MultiLineString", "Polygon", "MultiPolygon", "GeometryCollection", "Feature", "FeatureCollection"]
+
   public init(data: Data?, response: URLResponse?, error: NSError?) {
     self.data = data
     self.response = response
     self.error = error
     if let dictResponse = parseData(data) {
-      parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
+      if let type = dictResponse.value(forKey: "type") as? String {
+        if PeliasResponse.validTypes.contains(type) {
+          parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
+        }
+      } else {
+        let meta = dictResponse.object(forKey: "meta") as? NSDictionary
```

Comment:
Minor nit - i think as of Swift 3, NSDictionary is properly bridged now to Swift's `Dictionary` and supports subscripted access, so you can convert these to `dictResponse["meta"]` for example.  However I think to get it to work I think you have to specify parameterized types coming out of `parseData`. Might be too much work for this PR though?